### PR TITLE
chore: use testcontainers-go in redis receiver

### PR DESCRIPTION
**Pull request by mdelapenya**
**Description:** <Describe what has changed.>

This PR is the first one replacing the internal package for managing containers lifecycle in tests with testcontainers-go. In this case, we are updating the Redis receiver tests.

**Link to tracking Issue:** #9939

**Testing:** <Describe what testing was performed and which tests were added.>

I ran the integration tests for Redis with:

```
cd receiver/redisreceiver
go test -timeout 600s -v -run ^TestIntegration$ -tags integration
```

**Documentation:** No docs were updated

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
Thanks for the PR! Please sign the CLA

File: receiver/redisreceiver/go.mod [hunk 1]
```diff
@@ -4,37 +4,47 @@ go 1.18
 
 require (
 	github.com/go-redis/redis/v7 v7.4.1
-	github.com/open-telemetry/opentelemetry-collector-contrib/internal/containertest v0.64.0
 	github.com/stretchr/testify v1.8.1
+  github.com/testcontainers/testcontainers-go v0.15.0
```
[the remark this answers is not in the document]
Comment (mdelapenya) [reply receiver/redisreceiver/go.mod]: Done in ebee005, sorry about that. I manually resolved conflicts using GH's UI 🤦

**Comment by mdelapenya** [comment]
@codeboten @djaglowski I signed the CLA so I think this is ready to go!

Thanks in advance for your support!

**Review by TylerHelmuth — APPROVED** [review]
@mdelapenya thanks!

**Review by djaglowski — APPROVED** [review]
Thank you @mdelapenya!

**Review by codeboten — APPROVED** [review]
Thanks @mdelapenya! I've restarted the failed CI checks, i dont think they were related to this change.

**Comment by mdelapenya** [comment]
Great to know it! I was about to comment exactly that 😅  Hopefully the build passes and I can continue sending the updates in batches.

Thanks!

**Review by dmitryax — APPROVED** [review]

**Pull request merged**